### ISSUE TITLE
Your tools on a QNAP TS-251+ NAS (Hardware transcoding possible(?))

### ISSUE THREAD

**Issue by IchBinGROOT** [donmelton/other_video_transcoding#84]
Last week I started to rip my collection of DVDs (really, I do not have any BluRays) with MakeMKV, and then I started compressing them with Handbrake, but I really do not know if this is the best option as I am very new to this topic. The goal is to have the movies as mp4 files on my Qnap NAS, as it serves as a mediaserver in my network. I only have an old laptop with an optical drive for the rips, and a slightly newer one I can use for the transcoding (both running Windows 10), and it is incredibly slow. After some googling I found out that my QNAP TS-251+ is supposed to be able of hardware transcoding, so I would like to use this in order to be more efficient. As I am from Germany, I would also like to have both the german and the original audiotrack in the result file, as well as both german and english subtitles. MakeMKV recognizes 5.1 Dolby Surround Sound tracks on the DVDs, and by default, Handbrake converts them into stereo, I don't know how much difference it makes about the file size. My NAS only has 6TB of storage in RAID1, so about 3TB usable space.
Now I look for a way to be able to just rip the DVDs with the old laptop using MakeMKV, dump them into a specified folder on the NAS, and then have it transcoding all of those into mp4s (of cause without losing to much quality along the way). Does anybody do something similar and can help me build such a workflow? Does it even make sense, or should I rip the DVDs with the first laptop, copy the mkvs to the second, transcode them there (using Handbrake or some of your tools [any parameter suggestions?]) and then copy them to the NAS?
Thanks for your support.

**Comment by khaosx** [comment]
@IchBinGROOT - I've never tried to do manual transcoding on my QNAP, but I do currently run Plex on it with hardware transcoding enabled, and it seems to work fine, suggesting that what you're trying to do is theoretically possible. I suspect your biggest headache would be the limitations of the BusyBox OS that QNAP runs. If you can get the proper tool chain installed it is doable, but I can't provide a lot of help with that.

You could also look into running a Docker container with the appropriate tools already rolled into the container. I think we have some folks hanging around that might be able to give some insight onto that solution.

**Comment by IchBinGROOT** [comment]
Thanks a lot, hearing that it is possible feels good. I am not in a hurry, I don’t mind waiting for other people to comment. I have no experience with Docker, but if this is the solution, I am willing to try it, but I will probably need some help setting it up.

**Comment by ttyS0** [comment]
It looks like that model NAS has a [Celeron J1900 CPU](https://ark.intel.com/content/www/us/en/ark/products/78867/intel-celeron-processor-j1900-2m-cache-up-to-2-42-ghz.html) in it, which technically supports Intel QuickSync (QSV), but I'm not sure about either the quality or the performance of that iGPU. If you have an Intel CPU in pretty much any other system that is somewhat recent (last four years or so) you'd see significant improvements in transcode speed and probably better quality. 

For Plex it's probably fine. There's an assumption that most content would streamed via Direct Play (i.e. your network is fast enough to just stream the video as it is), and in the case of Plex needing to transcode something, it's only for that viewing session. When thinking about using `other-transcode`, the context is to create a new copy of an "original" that is as close to the original's quality as possible while also being smaller in size. In that context, I don't think I'd want to leverage that Celeron iGPU. 

While that CPU does have QSV support, the next question is whether the `/dev/dri` device is exposed to the Docker environment. I don't have a QNAP, so I don't know how much direct access one has to Docker or the underlying Linux environment. I do have a [QSV Docker image](https://github.com/ttyS0/docker-other-transcode). It requires mapping of the `/dev/dri` device, and volume mapping so `other-transcode` can find source files and write them out to the mapped volume. I'm very hesitant to suggest that your first Docker experience be from a QNAP UI.

tl;dr Technically that QNAP model's CPU supports Intel QSV. You would get better quality and performance from an Intel Core CPU released any time in the past four years. Alternatively if you have an Nvidia card of almost any kind, you'd also get better quality and performance from that, regardless of your CPU. 😎

**Comment by IchBinGROOT** [comment]
So you suggest that I just use MakeMKV to rip the DVDs, then use `other-transcode` on my main laptop, and then copy the finished file to my QNAP? Or would you rather buy an old(er) graphics card on Ebay or somewhere and build a dedicated transcoding pc?
And what about the other part of the question, is it possible to move multiple subtitles that I extracted from the DVD with MakeMKV to the new file? Or would you search for a subtitle file online in order to import that one into the finished movie? As I need multiple subtitles, burning them into the video is no option for me. And would you recommend to keep the Dolby 5.1 surround sound from the DVD, or just transcode that as well into a stereo sound?
Just to make things clear, I (or better, my family) own all the DVDs I want to transcode, but I'd also like to take the movies with me, when I'm not at home. In addition, just ripping the DVDs to a (or probably multiple) harddrive(s) and not transcoding them is not an option, as I really don't have that much space on any drive I've lying around. At the moment, we do not have a surround sound system, but chances are that I will have one in the future, so preferably I would like to keep the possibility to use the 5.1 track, without having to manually rip all the DVDs another time. How much of a difference in file size does it make?

**Comment by IchBinGROOT** [comment]
> While that CPU does have QSV support, the next question is whether the /dev/dri device is exposed to the Docker environment. I don't have a QNAP, so I don't know how much direct access one has to Docker or the underlying Linux environment. I do have a QSV Docker image. It requires mapping of the /dev/dri device, and volume mapping so other-transcode can find source files and write them out to the mapped volume. I'm very hesitant to suggest that your first Docker experience be from a QNAP UI.

Ok then, I think this is somewhat true. I definitely do not like the QNAP gui, but I also have no real experience with linux systems in general.

**Comment by weaverm** [comment]
A tool called MediaInfo ([https://en.wikipedia.org/wiki/MediaInfo](https://en.wikipedia.org/wiki/MediaInfo)) can help you understand the various bits of these files. For example, a movie I have on DVD and encoded into a h.264 file is 1.79 GiB overall, but the video track is 1.28 GiB, the 6 channel Dolby Digital audio track is 295 MiB, and the 2 channel AAC mix down is 109 MiB. In short, the video parts are much bigger than the audio parts.

In my view, the thing that really takes all of one's time is getting bits off an optical disc via MakeMKV since it requires more attention putting discs into a reader and making sure things are named appropriately. The encoding can easily happen while one sleeps. 

Having the 5.1 audio tracks in your files seems like a good trade off since it only costs a small amount of disk space but potentially saves a lot of time in the future.

**Comment by ttyS0** [comment]
> So you suggest that I just use MakeMKV to rip the DVDs, then use other-transcode on my main laptop, and then copy the finished file to my QNAP? Or would you rather buy an old(er) graphics card on Ebay or somewhere and build a dedicated transcoding pc?

Yep! If you see yourself doing a lot of transcoding because you're going to be accumulating more media, then a dedicated transcoding system would make sense. If you have a pile of media that you want to chug through, and are then basically done, your existing laptop should be fine, especially if it's relatively new. For a dedicated system you really don't even need a dedicated graphics card if you get a newish Intel CPU, as the QSV support in newer CPUs is quite good (and fast). 

> is it possible to move multiple subtitles that I extracted from the DVD with MakeMKV to the new file? ... And would you recommend to keep the Dolby 5.1 surround sound from the DVD, or just transcode that as well into a stereo sound?

My subtitle expertise is super minimal, but as @weaverm alluded to it's generally a good idea to keep as much information in your ripped file as possible. So, if you know you're going to want multiple subtitles, and they are available via the media proper (i.e. they're in the list of things that MakeMKV is going to rip for a title), then you should include them. The same goes for any audio tracks. 

When you transcode from that rip file, you can then choose whether you want to have the subtitles or audio tracks included in the transcoded file. Lets say some reason you don't want to include commentary tracks in your transcode. If you change your mind later, then you just need to run a transcode from your saved rip file. If you didn't include them in your rip file, then you have to go back to the original media and make another rip, which is always going to be a slow and tedious process. 

I think I may have buried the lede there. Basically, gather up all the information from the rip that you think you _might maybe_ want. Save the rip files, and then at transcoding time decide what you want to include. And yes, this means you'll probably start accumulating more hard drives. `This is the way.` 😎

**Comment by IchBinGROOT** [comment]
Ok, I understood that I need more storage space on my disks. Any suggestions? And as I am already asking for hardware suggestions, any recommendations for an external DVD drive that is capable of "LibreDrive Mode" for MakeMKV?

And @weaverm you put both the original 5.1 surround sound track and a stereo track into the final file? Could you just paste your settings (or your command parameters) that I have something to start experimenting with myself?

**Comment by ttyS0** [comment]
For optical drivesI I think [this MakeMKV forum post](https://www.makemkv.com/forum/viewtopic.php?t=19634) is one of the default goto places. A lot of folks seem to like the [Asus BW-16D1HT](https://amzn.to/3llVCK1), specifically. If you're just ripping DVDs, I also don't think it really matters. In fact, if you're just ripping non-4K BluRays, I don't think it really matters. I use a four year old Pioneer BDR-XD05S that's USB bus powered. It's literally had hundreds of BluRays go through it, and it just has the stock firmware on it. 

As far as audio options for `other-transcode`, you might want to check out the Wiki in this repo: https://github.com/donmelton/other_video_transcoding/wiki/Audio

**Comment by IchBinGROOT** [comment]
Okay then, thank you all very much for your help. This weekend I'm going to rip some more DVDs and test some parameters of your script, in order to see if everything works as expected. I think I will probably come back here to ask for some more help, but I will see.

**Comment by donmelton** [comment]
@IchBinGROOT Feel free to comment again here or open a new issue. Closing this now.
